boards/DCM/Test/Src/Test_StateMachine.cpp

```diff
@@ -399,4 +409,147 @@ TEST_F(
+
+    // Check regen doesn't turn on when both wheels are going faster than 5kph
+    // but one of the AIRs is open
+    App_CanRx_FSM_WHEEL_SPEED_SENSOR_SetSignal_LEFT_WHEEL_SPEED(
+        can_rx_interface, 5.1f);
+    App_CanRx_FSM_WHEEL_SPEED_SENSOR_SetSignal_RIGHT_WHEEL_SPEED(
+        can_rx_interface, 5.1f);
+
+    App_CanRx_BMS_AIR_STATES_SetSignal_AIR_NEGATIVE(
+        can_rx_interface, CANMSGS_BMS_AIR_STATES_AIR_NEGATIVE_OPEN_CHOICE);
+    App_CanRx_BMS_AIR_STATES_SetSignal_AIR_POSITIVE(
+        can_rx_interface, CANMSGS_BMS_AIR_STATES_AIR_POSITIVE_CLOSED_CHOICE);
+    LetTimePass(state_machine, 10);
+    EXPECT_TRUE(
+        App_CanTx_GetPeriodicSignal_TORQUE_REQUEST(can_tx_interface) >= 0);
+
+    App_CanRx_BMS_AIR_STATES_SetSignal_AIR_NEGATIVE(
+        can_rx_interface, CANMSGS_BMS_AIR_STATES_AIR_NEGATIVE_CLOSED_CHOICE);
+    App_CanRx_BMS_AIR_STATES_SetSignal_AIR_POSITIVE(
+        can_rx_interface, CANMSGS_BMS_AIR_STATES_AIR_POSITIVE_OPEN_CHOICE);
+    LetTimePass(state_machine, 10);
+    EXPECT_TRUE(
+        App_CanTx_GetPeriodicSignal_TORQUE_REQUEST(can_tx_interface) >= 0);
+
+    // Check regen turns on when both wheels are going faster than 5kph and AIRS
+    // are closed
+    App_CanRx_BMS_AIR_STATES_SetSignal_AIR_NEGATIVE(
+        can_rx_interface, CANMSGS_BMS_AIR_STATES_AIR_NEGATIVE_CLOSED_CHOICE);
+    App_CanRx_BMS_AIR_STATES_SetSignal_AIR_POSITIVE(
+        can_rx_interface, CANMSGS_BMS_AIR_STATES_AIR_POSITIVE_CLOSED_CHOICE);
+    LetTimePass(state_machine, 10);
+    EXPECT_TRUE(
+        App_CanTx_GetPeriodicSignal_TORQUE_REQUEST(can_tx_interface) < 0);
+}
+
+// DCM-19
+TEST_F(DcmStateMachineTest, torque_requests_faster_than_100Hz)
+{
+    float torque_request_with_regen_allowed;
+    float torque_request_without_regen_allowed;
+
+    SetInitialState(App_GetDriveState());
+
+    // Turn the DIM start switch on to prevent state transitions in
+    // the drive state.
+    EXPECT_TRUE(App_BuzzerSignals_IsOn(world));
+    App_CanRx_DIM_SWITCHES_SetSignal_START_SWITCH(
+        can_rx_interface, CANMSGS_DIM_SWITCHES_START_SWITCH_ON_CHOICE);
+    ASSERT_EQ(
+        App_GetDriveState(),
+        App_SharedStateMachine_GetCurrentState(state_machine));
+
+    // Check that no torque requests are sent when the accelerator pedal is not
+    // pressed
+    // ("The DCM may only request torque less than or equal to what the driver
+    // requested")
+    LetTimePass(state_machine, 10);
+    ASSERT_EQ(
+        0.0f, App_CanTx_GetPeriodicSignal_TORQUE_REQUEST(can_tx_interface));
+
+    // Check that positive torque requests are sent when the accelerator pedal
+    // is pressed
+    App_CanRx_FSM_PEDAL_POSITION_SetSignal_MAPPED_PEDAL_PERCENTAGE(
+        can_rx_interface, 23.2f);
+    LetTimePass(state_machine, 10);
+    torque_request_without_regen_allowed =
+        App_CanTx_GetPeriodicSignal_TORQUE_REQUEST(can_tx_interface);
+    EXPECT_TRUE(torque_request_without_regen_allowed > 0.0f);
```

Comment:
Will MAPPED_PEDAL_PERCENTAGE always be a value between 0 and 100f or should I add a check to make sure that torque requests don't increase when MAPPED_PEDAL_PERCENTAGE > 100?

Reply:
No need for the check here

Reply:
ok